lib/core/embedded-fonts/build.rs

```diff
@@ -0,0 +1,113 @@
+use std::{path, env, fs, io};
+use std::io::Write;
+
+// =====================
+// === FillMapRsFile ===
+// =====================
+
+/// Generated file with code filling embedded fonts map
+pub struct FillMapRsFile {
+    file : fs::File
+}
+
+impl FillMapRsFile {
+    fn create<P:AsRef<path::Path>>(path:P) -> io::Result<FillMapRsFile> {
+        let mut file = fs::File::create(path)?;
+        writeln!(file, "{{")?;
+        Ok(FillMapRsFile{ file })
+    }
+
+    fn add_font_inserting_line(&mut self, font_name:&str, font_file:&str) -> io::Result<()> {
+        writeln!(self.file,
+            "   font_data_by_name.insert(\"{font_name}\", include_bytes!(\"{font_file}\"));",
+            font_name = font_name,
+            font_file = font_file
+        )
+    }
+
+    fn close_block(&mut self) -> io::Result<()> {
+        writeln!(self.file, "}}")
+    }
+}
+
+// ====================
+// === DejaVu fonts ===
+// ====================
+
+mod deja_vu {
+    use crate::FillMapRsFile;
+
+    use std::path;
+    use basegl_build_utilities::GithubRelease;
+
+    pub const PACKAGE : GithubRelease = GithubRelease {
+        project_url : "https://github.com/dejavu-fonts/dejavu-fonts/",
+        version     : "version_2_37",
+        filename    : "dejavu-fonts-ttf-2.37.zip"
+    };
+
+    pub const PACKAGE_FONTS_PREFIX : &str = "dejavu-fonts-ttf-2.37/ttf";
+
+    pub fn font_file_from_font_name(font_name:&str) -> String {
+        return format!("{}.ttf", font_name);
+    }
+
+    pub fn extract_font(package_path:&path::Path, font_name:&str) {
+        let font_file            = font_file_from_font_name(font_name);
+        let font_in_package_path = format!("{}/{}",PACKAGE_FONTS_PREFIX,font_file);
+        let package_dir          = package_path.parent().unwrap();
+        let output_path          = package_dir.join(font_file);
+
+        let archive_file     = std::fs::File::open(package_path).unwrap();
+        let mut archive          = zip::ZipArchive::new(archive_file).unwrap();
+        let mut input_stream     = archive.by_name(font_in_package_path.as_str()).unwrap();
+        let mut output_stream    = std::fs::File::create(output_path).unwrap();
+        std::io::copy(&mut input_stream, &mut output_stream).unwrap();
+    }
+
+    pub const FONTS_TO_EXTRACT : &[&str] = &[
+        "DejaVuSans",
+        "DejaVuSans-ExtraLight",
+        "DejaVuSansMono",
+        "DejaVuSansMono-Bold",
+        "DejaVuSansMono-Oblique",
+        "DejaVuSansCondensed",
+        "DejaVuSerif",
+        "DejaVuSerifCondensed" ];
+
+    pub fn extract_all_fonts(package_path : &path::Path) {
+        for font_name in FONTS_TO_EXTRACT {
+            extract_font(package_path, font_name);
+        }
+    }
+
+    pub fn download_and_extract_all_fonts(out_dir : &path::Path) {
+        let package_path = out_dir.join(PACKAGE.filename);
+
+        PACKAGE.download(&out_dir);
+        extract_all_fonts(package_path.as_path());
```

Comment:
Well, actually I don't have problem with fast scrolling when the functions have only one line spacing, because I rather looks for indentation, and closing bracket `}` makes an additional empty line between functions. 

And the line spacing inside functions helps me sometimes, especially since I follow the coding guide - I make many _alias_ lets, which provide only nice naming, but no logic, therefore I'd like to see where these aliases end and begin the main logic of the function.

The good example is `extract_font` function you also commented above. There you have "a preparation section" where all the paths and names are built and named, then is "the actual stuff section", where we do the main logic. And when I read such funcions, I usually read the "actual stuff" section, looking on the first section only if some of aliases used are not clear to me.

Reply:
So, for now I removed some of empty lines - tried to leave to most one per function

Reply:
Super cool. I agree 100%.